Lib/typing.py

```diff
@@ -164,6 +162,22 @@
     'Unpack',
 ]
 
+class _LazyAnnotationLib:
+    def __init__(self) -> None:
+        self._annolib = None
+
+    def __getattr__(self, attr):
+        if self._annolib is None:
+            import annotationlib
+
+            self._annolib = annotationlib
+
+        value = getattr(self._annolib, attr)
+        setattr(self, attr, value)
+        return value
+
+_lazy_annotationlib = _LazyAnnotationLib()
```

Comment:
couldn't we do this more simply as

```suggestion
annotationlib = None

def _lazy_annotationlib():
    global annotationlib

    if annotationlib is None:
        import annotationlib
        
    return annotationlib
```

which would be similar to the approach taken in other modules:

https://github.com/python/cpython/blob/8681c97c0ec3c73332ba54a48b9c450192704cce/Lib/random.py#L155-L163

and then you could just use `_lazy_annotationlib()` instead of `_lazy_annotationlib` at the places that currently use `annotationlib`

Reply:
Note that my approach `setattr`s the attribute on the `_lazy_annotationlib` object, so it's likely faster than your approach, which introduces a function call overhead at every usage. However, we could do even better by replacing the `_lazy_annotationlib` global with the actual `annotationlib` module after it's imported, let me do that.